### PR TITLE
Private Events - Complete Projects

### PR DESCRIPTION
**Pull request by GabrielJSuarez**
- Create pull request for the Private Events project
- Created basic authentication with sessions
- Created models for users/events/attendances
- Created queries to divide past and upcoming events
- Added validations in views and databases

### CODE REVIEW

File: app/views/events/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    <li class="list-group-item font-weight-bold">
+      Attendance List!
+    </li>
+    <% @event.attendees.each do |person| %>
+      <li class="list-group-item">
+        <%= person.name %>
+      </li>
+    <% end %>
+  </ul>
+</div>
```
Comment (ghost) [inline app/views/events/show.html.erb]: There should be no logic (conditional statements and loops) in this file. Create them in View helpers `(app/helpers/*_helper.rb)` instead.

File: app/models/attendance.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Attendance < ApplicationRecord
+  belongs_to :attendee, class_name: 'User'
+  belongs_to :attended_event, class_name: 'Event'
+
+  validates :attendee_id, uniqueness: { scope: :attended_event_id, message: 'User is already attending this event' }
+end
```
Comment (ghost) [inline app/models/attendance.rb:6]: Add spec tests for the above associations and validations.

**Review by ghost — CHANGES_REQUESTED** [review]
# Status: Changes Required

### Hello team 😃,

**Great job on your project 💪. There are, however, a few fixes needed before the project can be approved.**

1). **A logged-in user should be able to attend an event whether it is a future event or past event.**

2). **Add a short description next to the project title in your readme, of what the project is about.**

3). **Add the following files/folders in `.gitignore` and remove them from your git repo: `.idea` and `.browserslistrc`.**

4). **In your readme Getting started manual you can instruct users to run` yarn install/ npm install` to install bootstrap dependencies since they are already listed in the `package.json`. `yarn add` is used for dependencies not listed in package.json.**

5). **See two more comments below.**

[Optional]

- You have misspelt sign up on your homepage.

- There is code that is displaying at the bottom of every event.

## Happy coding 🥂 

**PS: You can reach me on slack (Rose Wanjohi).**

**Review by voscarmv — APPROVED** [review]
# Project status: Approved ✨ 😎👍 💯 ✔️

- Good work applying all necessary changes.

### You may merge this branch and move forward ⛙🛣

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

**Pull request merged**